File: src/infra/transfer-repository.ts
import Transfer from '../models/transfer'
import Sqlitedb from './sqlitedb'

export default class TransferRepository {

    private db: Sqlitedb
    private tableName: string = 'transfers'

    constructor() {
        this.db = new Sqlitedb();
    }

    async getAll(): Promise<Transfer[]> {
        const entities = await this.db.getAll<Transfer>(this.tableName)
        return entities
    }

    async getById(id: number): Promise<Transfer | null> {
        const entity = await this.db.getById<Transfer>(this.tableName, id)
        return entity
    }

    async create(entity: Transfer): Promise<number> {
        const id = await this.db.create(this.tableName, entity)
        return id
    }

    async update(entity: Transfer): Promise<void> {
        await this.db.update(this.tableName, entity)
    }
}

File: Fin.Infrastructure/Repositories/TransferRepository.cs
using Fin.Domain.Entities;
using Fin.Infrastructure.Data;

namespace Fin.Infrastructure.Repositories
{
    public interface ITransferRepository
    {
        IQueryable<Transfer> GetAll();
        Transfer? Get(int id);
        Transfer Create(Transfer transfer);
        void Delete(Transfer transfer);
    }
    
    public class TransferRepository(FinDbContext context) : ITransferRepository
    {
        public IQueryable<Transfer> GetAll()
        {
            var entities = context.Transfers.Where(a => a.IsActive);
            return entities;
        }

        public Transfer? Get(int id)
        {
            var transfer = context.Transfers.Find(id);
            return transfer != null && transfer.IsActive ? transfer : null;
        }

        public Transfer Create(Transfer transfer)
        {
            ArgumentNullException.ThrowIfNull(transfer, nameof(transfer));

            transfer.IsActive = true;

            context.Transfers.Add(transfer);

            return transfer;
        }

        public void Delete(Transfer transfer)
        {
            transfer.IsActive = false;
        }
    }
}